Update compatible with Api v2

stevejcl opened the pull request:
New version compatible with API V2 to add to a new branch (apiV2) on the original project

acocalypso commented:
Hey currently testing this PR and I have the following issue while connecting to the Dwarf:

```
Unhandled Runtime Error
TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor

Source
src\components\setup\ConnectDwarf.tsx (114:8) @ checkConnection

  112 | const webSocketHandler = connectionCtx.socketIPDwarf
  113 |   ? connectionCtx.socketIPDwarf
> 114 |   : new WebSocketHandler(IPDwarf);
      |    ^
  115 | 
  116 | connectionCtx.setSocketIPDwarf(webSocketHandler);
  117 | 
Call Stack
HTMLUnknownElement.callCallback
node_modules\react-dom\cjs\react-dom.development.js (4164:0)
Object.invokeGuardedCallbackDev
node_modules\react-dom\cjs\react-dom.development.js (4213:0)
invokeGuardedCallback
node_modules\react-dom\cjs\react-dom.development.js (4277:0)
invokeGuardedCallbackAndCatchFirstError
node_modules\react-dom\cjs\react-dom.development.js (4291:0)
executeDispatch
node_modules\react-dom\cjs\react-dom.development.js (9041:0)
processDispatchQueueItemsInOrder
node_modules\react-dom\cjs\react-dom.development.js (9073:0)
processDispatchQueue
node_modules\react-dom\cjs\react-dom.development.js (9086:0)
dispatchEventsForPlugins
node_modules\react-dom\cjs\react-dom.development.js (9097:0)
eval
node_modules\react-dom\cjs\react-dom.development.js (9288:0)
batchedUpdates$1
node_modules\react-dom\cjs\react-dom.development.js (26140:0)
batchedUpdates
node_modules\react-dom\cjs\react-dom.development.js (3991:0)
dispatchEventForPluginEventSystem
node_modules\react-dom\cjs\react-dom.development.js (9287:0)
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay
node_modules\react-dom\cjs\react-dom.development.js (6465:0)
dispatchEvent
node_modules\react-dom\cjs\react-dom.development.js (6457:0)
dispatchDiscreteEvent
node_modules\react-dom\cjs\react-dom.development.js (6430:0)
```

Silky-Thrower commented:
![problem connecting](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/d88d9e4d-3cf7-4156-957c-92e1add47572)

Silky-Thrower commented:
![problem connecting](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/eb0109f7-54a1-434a-a87c-bff597ed0191)

Silky-Thrower commented:
The prebuild version with on localhost:8000 works normal
But only in slave mode... why do i have to be connected with my phone ?
![problem connecting-prebuild](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/f0fac3c0-2cf0-415d-b887-c88751aa39af)

acocalypso commented:
You need the dwarf to join your local network with sta mode.

Silky-Thrower commented:
> You need the dwarf to join your local network with sta mode.

i know and it is in sta mode, but i can't connect to it without connecting the phone first

stevejcl commented:
yes the first connection need Bluetooth, so it needs the mobile app even in STA Mode, I add this response from Dwarflab,  They will send me the Bluetooth API connection (I hope soon), so then we can directly connect to the dwarf.

acocalypso commented:
I tested it today an saw that exposures up to 30 sec can be set.

I tried it with 20s exposures and 30 frames but session started with 999 1s exposures.

stevejcl commented:
yes I have made  a test, as the configuration is a JSON file on the dwarf, I try to change it and test if it accept bigger value, (in fact the darf receives an Index value not the direct value), but it's not working, you receive a bad parameter response, so I revert back in the last version.

acocalypso commented:
I got an ok response and the session started but instead of the configured subs amount and exposure time, the dwarf started with 999 1s exposures.

stevejcl commented:
Ok, got the Bluetooth API, now I can start the STA mode without the APP,, for the moment for the moment I just made a test with an hard coded packet, but it's working

acocalypso commented:
I checkout the new changes and I'm not able to start it now:

```
./src/components/ConnectCamera.tsx:3:0
Module not found: Can't resolve '@/lib/dwarfii_api'
  1 | import { useContext, useState } from "react";
  2 | 
> 3 | import {
  4 |   wsURL,
  5 |   statusTelephotoCmd,
  6 |   statusWideangleCmd,

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./src/pages/setup-scope.tsx
```

acocalypso commented:
> I checkout the new changes and I'm not able to start it now:
> 
> ```
> ./src/components/ConnectCamera.tsx:3:0
> Module not found: Can't resolve '@/lib/dwarfii_api'
>   1 | import { useContext, useState } from "react";
>   2 | 
> > 3 | import {
>   4 |   wsURL,
>   5 |   statusTelephotoCmd,
>   6 |   statusWideangleCmd,
> 
> https://nextjs.org/docs/messages/module-not-found
> 
> Import trace for requested module:
> ./src/pages/setup-scope.tsx
> ```

ignore that I cloned your working repository and its working there, It seems there was an issue with checking out the PR

stevejcl commented:
> > I checkout the new changes and I'm not able to start it now:
> > ```
> > ./src/components/ConnectCamera.tsx:3:0
> > Module not found: Can't resolve '@/lib/dwarfii_api'
> >   1 | import { useContext, useState } from "react";
> >   2 | 
> > > 3 | import {
> >   4 |   wsURL,
> >   5 |   statusTelephotoCmd,
> >   6 |   statusWideangleCmd,
> > 
> > https://nextjs.org/docs/messages/module-not-found
> > 
> > Import trace for requested module:
> > ./src/pages/setup-scope.tsx
> > ```
> 
> ignore that I cloned your working repository and its working there, It seems there was an issue with checking out the PR

Yes I left the link to this repo in my github to be able to integrate it here with pull request, this could cause problems

acocalypso commented:
Alright, I just pulled the latest commits and I'm not able to connect to the DWARF now.
I don't have a compatible bluetooth usb dongle yet but I followed the standart procedure.

I typed in my ip from the Dwarf and hit connect:

```
Unhandled Runtime Error
TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor

Source
src\components\setup\ConnectDwarf.tsx (48:8) @ checkConnection

  46 | const webSocketHandler = connectionCtx.socketIPDwarf
  47 |   ? connectionCtx.socketIPDwarf
> 48 |   : new WebSocketHandler(IPDwarf);
     |    ^
  49 | 
  50 | connectionCtx.setSocketIPDwarf(webSocketHandler);
  51 | 
```

Do you have any discord server or similiar to discuss issues etc or should we stick to github?
I'm glad to help

i can see the same issue that was reported 3 weeks ago.

acocalypso commented:
I opend up VS Code to double check:

Module '"dwarfii_api"' has no exported member 'Dwarfii_Api'

Where is the module defined?

Silky-Thrower commented:
> Alright, I just pulled the latest commits and I'm not able to connect to the DWARF now. I don't have a compatible bluetooth usb dongle yet but I followed the standart procedure.
> 
> I typed in my ip from the Dwarf and hit connect:
> 
> ```
> Unhandled Runtime Error
> TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> 
> Source
> src\components\setup\ConnectDwarf.tsx (48:8) @ checkConnection
> 
>   46 | const webSocketHandler = connectionCtx.socketIPDwarf
>   47 |   ? connectionCtx.socketIPDwarf
> > 48 |   : new WebSocketHandler(IPDwarf);
>      |    ^
>   49 | 
>   50 | connectionCtx.setSocketIPDwarf(webSocketHandler);
>   51 | 
> ```
> 
> Do you have any discord server or similiar to discuss issues etc or should we stick to github? I'm glad to help
> 
> i can see the same issue that was reported 3 weeks ago.

indeed, i have the same problem. 
 Unhandled Runtime Error
> TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> 
also on visual code this is always marked red ....
import {
    Dwarfii_Api,
    messageCameraTeleGetSystemWorkingState,
    messageCameraTeleOpenCamera,
    messageCameraWideOpenCamera,
    WebSocketHandler,
} from "dwarfii_api";

Silky-Thrower commented:
![bluetooth](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/4cd741e4-c2dd-4610-a103-0a6bcdf8980d)

acocalypso commented:
your released test build is working fine
https://github.com/stevejcl/dwarfii-stellarium-goto/releases/tag/v2.0.3a

@stevejcl what is the quality settings about? There is no explanation available in the camera information popup

stevejcl commented:
> > Alright, I just pulled the latest commits and I'm not able to connect to the DWARF now. I don't have a compatible bluetooth usb dongle yet but I followed the standart procedure.
> > I typed in my ip from the Dwarf and hit connect:
> > ```
> > Unhandled Runtime Error
> > TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> > 
> > Source
> > src\components\setup\ConnectDwarf.tsx (48:8) @ checkConnection
> > 
> >   46 | const webSocketHandler = connectionCtx.socketIPDwarf
> >   47 |   ? connectionCtx.socketIPDwarf
> > > 48 |   : new WebSocketHandler(IPDwarf);
> >      |    ^
> >   49 | 
> >   50 | connectionCtx.setSocketIPDwarf(webSocketHandler);
> >   51 | 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Do you have any discord server or similiar to discuss issues etc or should we stick to github? I'm glad to help
> > i can see the same issue that was reported 3 weeks ago.
> 
> indeed, i have the same problem. Unhandled Runtime Error
> 
> > TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> 
> also on visual code this is always marked red .... import { Dwarfii_Api, messageCameraTeleGetSystemWorkingState, messageCameraTeleOpenCamera, messageCameraWideOpenCamera, WebSocketHandler, } from "dwarfii_api";

You need my version of dwarfii_api on my git hub (https://github.com/stevejcl/dwarfii_api)  and copy the files to the ./node_modules/dwarfii_api rerpertory

acocalypso commented:
> > > Alright, I just pulled the latest commits and I'm not able to connect to the DWARF now. I don't have a compatible bluetooth usb dongle yet but I followed the standart procedure.
> > > I typed in my ip from the Dwarf and hit connect:
> > > ```
> > > Unhandled Runtime Error
> > > TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> > > 
> > > Source
> > > src\components\setup\ConnectDwarf.tsx (48:8) @ checkConnection
> > > 
> > >   46 | const webSocketHandler = connectionCtx.socketIPDwarf
> > >   47 |   ? connectionCtx.socketIPDwarf
> > > > 48 |   : new WebSocketHandler(IPDwarf);
> > >      |    ^
> > >   49 | 
> > >   50 | connectionCtx.setSocketIPDwarf(webSocketHandler);
> > >   51 | 
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Do you have any discord server or similiar to discuss issues etc or should we stick to github? I'm glad to help
> > > i can see the same issue that was reported 3 weeks ago.
> > 
> > 
> > indeed, i have the same problem. Unhandled Runtime Error
> > > TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> > 
> > 
> > also on visual code this is always marked red .... import { Dwarfii_Api, messageCameraTeleGetSystemWorkingState, messageCameraTeleOpenCamera, messageCameraWideOpenCamera, WebSocketHandler, } from "dwarfii_api";
> 
> You need my version of dwarfii_api on my git hub (https://github.com/stevejcl/dwarfii_api) and copy the files to the ./node_modules/dwarfii_api rerpertory

Thanks that explains a lot :D maybe you want to link it as a submodule?

stevejcl commented:
> > > > Alright, I just pulled the latest commits and I'm not able to connect to the DWARF now. I don't have a compatible bluetooth usb dongle yet but I followed the standart procedure.
> > > > I typed in my ip from the Dwarf and hit connect:
> > > > ```
> > > > Unhandled Runtime Error
> > > > TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> > > > 
> > > > Source
> > > > src\components\setup\ConnectDwarf.tsx (48:8) @ checkConnection
> > > > 
> > > >   46 | const webSocketHandler = connectionCtx.socketIPDwarf
> > > >   47 |   ? connectionCtx.socketIPDwarf
> > > > > 48 |   : new WebSocketHandler(IPDwarf);
> > > >      |    ^
> > > >   49 | 
> > > >   50 | connectionCtx.setSocketIPDwarf(webSocketHandler);
> > > >   51 | 
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Do you have any discord server or similiar to discuss issues etc or should we stick to github? I'm glad to help
> > > > i can see the same issue that was reported 3 weeks ago.
> > > 
> > > 
> > > indeed, i have the same problem. Unhandled Runtime Error
> > > > TypeError: dwarfii_api__WEBPACK_IMPORTED_MODULE_2__.WebSocketHandler is not a constructor
> > > 
> > > 
> > > also on visual code this is always marked red .... import { Dwarfii_Api, messageCameraTeleGetSystemWorkingState, messageCameraTeleOpenCamera, messageCameraWideOpenCamera, WebSocketHandler, } from "dwarfii_api";
> > 
> > 
> > You need my version of dwarfii_api on my git hub (https://github.com/stevejcl/dwarfii_api) and copy the files to the ./node_modules/dwarfii_api rerpertory
> 
> Thanks that explains a lot :D maybe you want to link it as a submodule?

Yes, I keep the old link to do the pull request to this repository, I am in this discord : https://discord.com/channels/1172554961704468572/1173899166850760724 and I have a developpement channel but I am feeling alone ...
https://discord.com/channels/1172554961704468572/1188219781271269529

acocalypso commented:
If you sent me a Discord Invite I can join =)

acocalypso commented:
for anyone else who would like to test it:
Installation of dwarfii_api module:

`npm install "https://github.com/stevejcl/dwarfii_api#apiV2" --save`

stevejcl commented:
> If you sent me a Discord Invite I can join =)

I am not the owner, send a PM to Alan Burgess https://www.facebook.com/groups/659927045624980/user/100058518950867 on messenger

stevejcl commented:
> your released test build is working fine https://github.com/stevejcl/dwarfii-stellarium-goto/releases/tag/v2.0.3a
> 
> @stevejcl what is the quality settings about? There is no explanation available in the camera information popup

I thing it the quality of the image preview on the mobile APP or the image get with http, not sure!

stevejcl commented:
> your released test build is working fine https://github.com/stevejcl/dwarfii-stellarium-goto/releases/tag/v2.0.3a
> 
> @stevejcl what is the quality settings about? There is no explanation available in the camera information popup

As I have 'not clear sky, what kind of test have you done ?

acocalypso commented:
> > your released test build is working fine https://github.com/stevejcl/dwarfii-stellarium-goto/releases/tag/v2.0.3a
> > @stevejcl what is the quality settings about? There is no explanation available in the camera information popup
> 
> As I have 'not clear sky, what kind of test have you done ?

I just did some basic tests, like functionality,  loading coords from Stellarium, testing all options.
I might be able to further test a full session tonight. in ~10h

Silky-Thrower commented:
> for anyone else who would like to test it: Installation of dwarfii_api module:
> 
> `npm install "https://github.com/stevejcl/dwarfii_api#apiV2" --save`

not working ....
i copied the whole api repo from stevejcl and i still have the same marked errors in vs 2019

acocalypso commented:
@Silky-Thrower all you have to do after running npm intall is installing the addional module from github with the command.

it should look like this:

```
npm install
npm WARN deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
npm WARN deprecated domexception@4.0.0: Use your platform's native DOMException instead

added 751 packages, and audited 752 packages in 24s

143 packages are looking for funding
  run `npm fund` for details

3 vulnerabilities (1 low, 2 moderate)

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.

npm install "https://github.com/stevejcl/dwarfii_api#apiV2" --save

up to date, audited 752 packages in 8s

143 packages are looking for funding
  run `npm fund` for details

3 vulnerabilities (1 low, 2 moderate)

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
```

Silky-Thrower commented:
> npm install "https://github.com/stevejcl/dwarfii_api#apiV2" --save

E:\Dwarf II\Dwarf II Stellarium - API2.0>npm install "https://github.com/stevejcl/dwarfii_api#apiV2" --save
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -4058
npm ERR! enoent An unknown git error occurred
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\silky\AppData\Local\npm-cache\_logs\2024-03-14T09_43_51_068Z-debug-0.log

E:\Dwarf II\Dwarf II Stellarium - API2.0>

stevejcl commented:
> npm install "https://github.com/stevejcl/dwarfii_api#apiV2" --save

try npm install username/repo#branchName --save

npm install stevejcl/dwarfii_api#apiV2 --save

I test it it's working

Silky-Thrower commented:
> > npm install "https://github.com/stevejcl/dwarfii_api#apiV2" --save
> 
> try npm install username/repo#branchName --save
> 
> npm install stevejcl/dwarfii_api#apiV2 --save
> 
> I test it it's working

what can i say... it's not working? 
every command gives me the same error
i even downgraded my node version from 20.11.1 to 20.9.0 and 18.17.1 ... . nothing works :/

acocalypso commented:
the error indicates an issue with your git binary.

can you run the following command from cmd (in case your are using windows)?

git --version

git version 2.44.0.windows.1

Silky-Thrower commented:
> the error indicates an issue with your git binary.
> 
> can you run the following command from cmd (in case your are using windows)?
> 
> git --version
> 
> git version 2.44.0.windows.1

That was the problem, now all the red markings are gone 
This should be in the install notes :)

Silky-Thrower commented:
Thx a million times :) 


![finally](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/85ada430-3a55-4202-bc2f-1f28da5f4e5f)

stevejcl commented:
> Thx a million times :)
> 
> ![finally](https://private-user-images.githubusercontent.com/154828328/312804836-85ada430-3a55-4202-bc2f-1f28da5f4e5f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MTUzMDYsIm5iZiI6MTcxMDQxNTAwNiwicGF0aCI6Ii8xNTQ4MjgzMjgvMzEyODA0ODM2LTg1YWRhNDMwLTNhNTUtNDIwMi1iYzJmLTFmMjhkYTVmNGU1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxNFQxMTE2NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYTA4Y2QxNjZiMWVjNzVlM2Q4ZDcyM2JmY2I3Y2VmOGE5YmNlNDhhZTJlNzc3NWViNmUwMGUxNGFkZmUyYzIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1YR9_AHllIfUC3DJqWA7yEb9iO56-pfB0XXHwUpfrFE)

What do you have changed to get this wall paper ?

Silky-Thrower commented:
> > Thx a million times :)
> > ![finally](https://private-user-images.githubusercontent.com/154828328/312804836-85ada430-3a55-4202-bc2f-1f28da5f4e5f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MTUzMDYsIm5iZiI6MTcxMDQxNTAwNiwicGF0aCI6Ii8xNTQ4MjgzMjgvMzEyODA0ODM2LTg1YWRhNDMwLTNhNTUtNDIwMi1iYzJmLTFmMjhkYTVmNGU1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxNFQxMTE2NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYTA4Y2QxNjZiMWVjNzVlM2Q4ZDcyM2JmY2I3Y2VmOGE5YmNlNDhhZTJlNzc3NWViNmUwMGUxNGFkZmUyYzIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1YR9_AHllIfUC3DJqWA7yEb9iO56-pfB0XXHwUpfrFE)
> 
> What do you have changed to get this wall paper ?

I build a whole site around it,  with info about stars, planets, constellations... 
Now i'm trying to include wheater and a clear outside data. 

![pic1](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/fd50c151-c457-472f-af0e-0177d0403ff7)
![pic2](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/8fc294a4-9daf-4509-af46-a3efe37b592c)
![pic3](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/8ef61f92-e1f0-4e61-bd35-2f71a536598a)
![pic4](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/154828328/3f80ff7e-624a-4b03-9f6e-5a5e77c02c5d)

acocalypso commented:
I love to work with desktop applications instead of webapps. Therefore I'm working on a desktop application for that.

![image](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/2846629/837fcb92-c59b-4e19-9428-d6b28457d0a8)

Silky-Thrower commented:
All combined would be an awesome application. 🤔

acocalypso commented:
> All combined would be an awesome application. 🤔

if @stevejcl would agree to this. I would support this absolutly. I could also offer to provide a github action to build and publish the packages for windows, linux and mac?

stevejcl commented:
I add this : 
![image](https://github.com/DwarfTelescopeUsers/dwarfii-stellarium-goto/assets/9412766/67dd733b-9c2f-4782-92f3-8e3d39ce1e7b)

to show only the visible objects depends on the category selected too

Silky-Thrower commented:
Is it possible for you @stevejcl to create a new repo? B ecause i think the merge will never happen. Also is there a way to communicate to you without spamming github?

acocalypso commented:
A new development branch would be good.

stevejcl commented:
on my github

acocalypso commented:
https://discord.com/invite/WZ5XxUJX

Feel free to join and discuss

acocalypso committed:
Merge branch 'apiV2' of https://github.com/acocalypso/dwarfii-stellarium-goto into apiV2

acocalypso committed:
format fix

acocalypso committed:
fix intend

Silky-Thrower committed:
Fix spinner

Delete spinner

Silky-Thrower committed:
Merge pull request #132 from Silky-Thrower/apiV2

Fix spinner

acocalypso committed:
Merge branch 'apiV2' into apiV2

acocalypso committed:
Merge pull request #48 from stevejcl/apiV2

sync

acocalypso committed:
small adjustment

acocalypso committed:
Merge branch 'apiV2' of https://github.com/acocalypso/dwarfii-stellarium-goto into apiV2

acocalypso committed:
fix dependencies

acocalypso committed:
fix eslint

acocalypso committed:
format fix

acocalypso committed:
Merge pull request #133 from acocalypso/apiV2

Fix dependencies

stevejcl committed:
rename variables to avoid compil error

acocalypso committed:
make nasaapikey persistent

acocalypso committed:
ignore lint until it breaks

acocalypso committed:
Merge branch 'apiV2' into apiV2

acocalypso committed:
Merge pull request #49 from stevejcl/apiV2

sync

acocalypso committed:
Merge pull request #134 from acocalypso/apiV2

Merge pull request #49 from stevejcl/apiV2

acocalypso committed:
fix footer

acocalypso committed:
Merge branch 'stevejcl:apiV2' into apiV2

acocalypso committed:
Merge pull request #135 from acocalypso/apiV2

fix footer size

stevejcl committed:
adjust Message Logs position

Silky-Thrower committed:
css fix

css fix calibration

Silky-Thrower committed:
Merge pull request #136 from Silky-Thrower/apiV2

css fix

Silky-Thrower committed:
Asteroids Goto Stellarium

Silky-Thrower committed:
Merge pull request #137 from Silky-Thrower/apiV2

Asteroids Goto Stellarium

acocalypso committed:
rework translation system based on json and i18n-ally

acocalypso committed:
Merge pull request #138 from acocalypso/apiV2

rework translation system based on json and i18n-ally

acocalypso committed:
Move Polaralignment logic from witmotion

acocalypso committed:
Merge branch 'stevejcl:apiV2' into apiV2

acocalypso committed:
Merge pull request #139 from acocalypso/apiV2

Move Polar Alignment Logic from Witmotion Sensor

acocalypso committed:
fix choice reader, enhance debug information

acocalypso committed:
Merge branch 'apiV2' of https://github.com/acocalypso/dwarfii-stellarium-goto into apiV2

acocalypso committed:
Merge branch 'stevejcl:apiV2' into apiV2

acocalypso committed:
Merge pull request #140 from acocalypso/apiV2

fix witmotion choice reader

Silky-Thrower committed:
Css Fix Polar

css fix polar

Silky-Thrower committed:
Merge pull request #141 from Silky-Thrower/apiV2

Css Fix Polar

Silky-Thrower committed:
Css

Silky-Thrower committed:
Merge branch 'stevejcl:apiV2' into apiV2

Silky-Thrower committed:
Merge pull request #142 from Silky-Thrower/apiV2

Css

Silky-Thrower committed:
Css add graphs

add graphs

Silky-Thrower committed:
Merge branch 'stevejcl:apiV2' into apiV2

Silky-Thrower committed:
Merge pull request #143 from Silky-Thrower/apiV2

Add Graphs css

stevejcl committed:
Modify Asteroid pages to make Goto working

Also format:fix and compilation OK

acocalypso committed:
add overlay for polar alignment

acocalypso committed:
Merge pull request #144 from acocalypso/apiV2

add overlay for polar alignment

acocalypso committed:
added chinese simplified

acocalypso committed:
adjusted nativ language

acocalypso committed:
format:fix

acocalypso committed:
Merge branch 'stevejcl:apiV2' into apiV2

acocalypso committed:
Merge pull request #145 from acocalypso/apiV2

chinese simplified support

Silky-Thrower committed:
css fix

css fix

Silky-Thrower committed:
Merge pull request #146 from Silky-Thrower/apiV2

css fix

Silky-Thrower committed:
Github corner

Silky-Thrower committed:
Merge pull request #147 from Silky-Thrower/apiV2

Github corner

acocalypso committed:
improve tiff handling

acocalypso committed:
Merge pull request #148 from acocalypso/apiV2

improve tiff handling

acocalypso committed:
format fix

acocalypso committed:
ignore lint

acocalypso committed:
Version 2.4.2

acocalypso committed:
Merge pull request #149 from acocalypso/apiV2

Version 2.4.2

acocalypso committed:
2.4.3 fix

acocalypso committed:
Merge branch 'stevejcl:apiV2' into apiV2

acocalypso committed:
Merge pull request #150 from acocalypso/apiV2

4.2.3 fix auto full screen

acocalypso committed:
remvoe not used keys

acocalypso committed:
Merge branch 'apiV2' of https://github.com/acocalypso/dwarfii-stellarium-goto into apiV2

acocalypso committed:
Merge branch 'stevejcl:apiV2' into apiV2

acocalypso committed:
Merge pull request #151 from acocalypso/apiV2

remove unsed keys from translation files

stevejcl committed:
add Missing Translations

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

stevejcl committed:
Add Português Translation

stevejcl committed:
Update CameraAddOn.tsx

correction for active button for stopping timelapse

acocalypso committed:
Merge pull request #50 from stevejcl/apiV2

sync

acocalypso committed:
Update package.json

acocalypso committed:
Update tauri.conf.json

acocalypso committed:
Update Footer.tsx

acocalypso committed:
Merge branch 'stevejcl:apiV2' into apiV2

acocalypso committed:
adjust rust toolchain

acocalypso committed:
bump rust toolchain version

acocalypso committed:
Merge pull request #152 from acocalypso/apiV2

2.4.4 Update

stevejcl committed:
add Dwarf 3 support

First test to prepare for D3, the dwarf version is found upon connection, and the correct config is loaded (for exposure / etc). Need to add filters options and mode capabilities for D3 : wide angle video support and see how we can integrate wide astrophotograpy

stevejcl committed:
add support for Dwarf 3

Version 2.4.5
Add exchange view for cameras

Preliminary support for Dwarf 3

stevejcl committed:
Update CameraAddOn.tsx

stevejcl committed:
Update package.json

update of joystick module breaks code

stevejcl committed:
Update tauri.conf.json

stevejcl committed:
Update Footer.tsx

Version 2.4.5

acocalypso committed:
update locales and readme

acocalypso committed:
add support for D3 Session Images & format fix

acocalypso committed:
Merge pull request #153 from acocalypso/apiV2

D3 adjustments

stevejcl committed:
correction for getting thumbnail images

stevejcl committed:
quality is specific to Dwarf II

stevejcl committed:
Update bluetooth to connect to Dwarf 3 too

acocalypso committed:
fix D3 ble connection where 4 bytes are sent

acocalypso committed:
Merge pull request #154 from acocalypso/apiV2

fix D3 ble connection where 4 bytes are sent

stevejcl committed:
set Dwarf Name and ID with bluetooth connection

stevejcl committed:
AI enhance not yet set by API

stevejcl committed:
Check if AiEnhance is defined before record it

it seems the D3 doesn(t return the AiEnhance value

stevejcl committed:
bluetooth connection for Dwarf 3

Bluetooth Dwarf 3 connection : use memory data in case config give Dwarf Hotstop data instead STA data

Get type of dwarf  from config files if bluetooth not used.

stevejcl committed:
Update astro_utils.ts

Correction for undefined data

stevejcl committed:
Update goto_utils.ts

Modification for positionning the Dwarf 3  in Polar Align

stevejcl committed:
format:fix

stevejcl committed:
Update Dwarfium with Wide Astro imaging for D3

Add Wide Astro Imaging for D3
Add Option to Connect  with another Dwarf ID to avoid conflict with astro session Tools.
Enable Camera view for D2 in Slave Mode .

stevejcl committed:
Add Dwarf 3 video streaming

Using a third party tools Mediamtx to stream to the web UI

Config file in data/mediamtx directory

stevejcl committed:
error compilation corrections

stevejcl committed:
add on for Dwarfium / MediaMTX

Add start command to launch Dwarfium and MediaMTX

Test before automatic build process

stevejcl committed:
add a exe for start up the batch file

stevejcl committed:
Update tauri.conf.json

Test install specific files

stevejcl committed:
Update tauri.conf.json

stevejcl committed:
Update tauri.conf.json

test windows

stevejcl committed:
Revert "Update tauri.conf.json"

This reverts commit 00ff95d30f0f2ae0a5557569b956c5fd7fa89615.

stevejcl committed:
revert changes

add changes in action files

stevejcl committed:
Update build.yml

stevejcl committed:
Update connect_utils.ts

not need

stevejcl committed:
Update Connect and getConfig Data function

stevejcl committed:
Update DwarfCameras.tsx

Optimize image to see them in production : needed to see correct size for streams

stevejcl committed:
Corrections for Images and Camera view

stevejcl committed:
New Test Version

Rewrite the Camera management
Need to use the MediaMtx software to enable streaming for D3 (exposure less than 1s)
Add Polar Align tool from Dwarflab.

stevejcl committed:
Update goto_utils.ts

correction for reading results of EQ Solving

stevejcl committed:
Test Version to resolve CORS problem

But use API routes that is not compatible with Tauri Build (/src/api/proxy.ts)
So Add a proxy test  in src/lib/proxyclient.ts , this will need a proxy server or Tauri can do that ??
=> See getDeviceInfoProxyRequest in /src/lib/get_dwarf_type.ts for test the proxy
Need to change .env.production to change variable
Also need a npm run  export to get the static web sites without routes

stevejcl committed:
Update ConnectDwarfSTA.tsx

Add wifi SSID and Password input, needed for Dwarf 3

stevejcl committed:
Update en.json

add language for home wifi input

stevejcl committed:
add proxy server for Dwarfium

stevejcl committed:
Update Tauri Build to add external bin files

- add Bluetooth STA info
- correction EQ Solving
- add more config build option
- add DockerFile to build Docker
- Tauri Build to launch Proxy and mediamtx for streaming D3 camera

stevejcl committed:
update github workflow

stevejcl committed:
Update build.yml

stevejcl committed:
verify tauri installed version

stevejcl committed:
test release version

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
test github action

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Version 4.5.0

Add Streams for Dwarf 3
Can setup Ip without bluetooth

stevejcl committed:
Update build.yml

stevejcl committed:
update github action

stevejcl committed:
Update test_artifacts.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update test_artifacts.yml

stevejcl committed:
Update test_artifacts.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Remove quality Preview as mandatory

stevejcl committed:
Change install dir for mediamtx config files

Change install dir for mediamtx config file
Add Dwarf Id in status bar

stevejcl committed:
change sub process launch

the process are stopped when exit app

stevejcl committed:
Update tauri.conf.json

add correct path for ressources and validator

Silky-Thrower committed:
Themesettings adjustments

css

Silky-Thrower committed:
Merge pull request #157 from Silky-Thrower/apiV2

Themesettings adjustments

Silky-Thrower committed:
Connection ID CSS

alignment  css

Silky-Thrower committed:
Merge branch 'stevejcl:apiV2' into apiV2

Silky-Thrower committed:
Merge pull request #158 from Silky-Thrower/apiV2

Connection ID CSS

Silky-Thrower committed:
Css Hamburger Menu

Adjustments

Silky-Thrower committed:
Merge pull request #159 from Silky-Thrower/apiV2

Css Hamburger Menu

stevejcl committed:
Updated Version

- use only version in package json to set version number, tauri conf will be updated during build
- set new port for DwarfiumProxy

stevejcl committed:
add Focus Info

The zip version can be now reached on the local network or even from outside with a tools like Tailscale
The location detection will not work, and the session data can(t downloaded in this case.

stevejcl committed:
Update tauri.conf.json

stevejcl committed:
correction for Tauri

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

stevejcl committed:
Add a dev command to start mediamtx

Silky-Thrower committed:
New function

Photo Editor Beta
update MoonPhase

Silky-Thrower committed:
Update PhotoEditor

Fix Social Button

stevejcl committed:
add Bluetooth Direct Connection for next dev

Format fix
Add proxy image upload for next dev
Need to correct a cross origin error in photo editor for zip build

stevejcl committed:
Update PhotoEditor.tsx

correction for crossorigin error

stevejcl committed:
Update README.md

stevejcl committed:
add multi configuration Environment for Proxy and Server

Add scripts to make  Server Standalone version
Add scripts to make  Proxy and Tools Standalone version
Add server Status Info
Add Help Message Optionnal in Setup Page
Add ImageEditor Translations
Correction for MoonePhase

stevejcl committed:
correction compil errors

stevejcl committed:
Update fr.json

stevejcl committed:
update translations

Silky-Thrower committed:
Menu

stevejcl committed:
Changes for Web Version compatibility on Smartphone

Prepare also Https version.

stevejcl committed:
format fix

stevejcl committed:
Update package.json

Silky-Thrower committed:
Menu fix

stevejcl committed:
Modify css and positionning for mobile

stevejcl committed:
Update Footer.tsx

For Dwarf Message stay visible on small screen

stevejcl committed:
Update Footer.tsx

Silky-Thrower committed:
MosaicPlanner

Add a function to create a mosaic with a downloaded json-file

Silky-Thrower committed:
Merge branch 'stevejcl:apiV2' into apiV2

Silky-Thrower committed:
fix css

Silky-Thrower committed:
Function Astro Session

Silky-Thrower committed:
format fix

Silky-Thrower committed:
fix

Silky-Thrower committed:
Live stream

stevejcl committed:
add HTTPS support

add tools to create SSL certificates
Support https Dwarfium Server / Vercel

stevejcl committed:
Add manual edit for Local Proxy IP

stevejcl committed:
correction when no local Ip found

Silky-Thrower committed:
Merge branch 'apiV2' into apiV2

Silky-Thrower committed:
Update nl.json

Silky-Thrower committed:
Merge pull request #1 from stevejcl/apiV2

Added mosaic, scheduler ui, live stream BETA

stevejcl committed:
Update Certifcates creation for Android

Correction for connection
Correction for exec : bluetooth and stellarium config

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

Silky-Thrower committed:
Merge branch 'stevejcl:apiV2' into apiV2

stevejcl committed:
correction for Streamings

To Do : correction of Status Bar Info (battery / Temperature)

Silky-Thrower committed:
Merge branch 'stevejcl:apiV2' into apiV2

Silky-Thrower committed:
UI Update

Weather update
css mobile fix

Silky-Thrower committed:
Merge pull request #163 from Silky-Thrower/apiV2

Ui update 
Weather 
mobile css fix

Silky-Thrower committed:
Adjustments Css

Clouds and weather update

Silky-Thrower committed:
Merge branch 'stevejcl:apiV2' into apiV2

Silky-Thrower committed:
Merge pull request #164 from Silky-Thrower/apiV2

Adjustments Css

stevejcl committed:
Correction for timeout error in streaming

detect if proxy is running on https for mediamtx and use mediamtx config file accordingly

stevejcl committed:
Corrections for Proxy - Server

add OBS studio protocol with Proxy

stevejcl committed:
Update createSSLcert.ts

Working on mobile, need to install manually the CA certificate

stevejcl committed:
Correction for Camera Conection outside Local Lan

stevejcl committed:
Update get_proxy_url.ts

error in Url

stevejcl committed:
Allows self-signed certificates on server

stevejcl committed:
Improve Proxy Server

- Finalize Certificates
- Improve Proxy Request
- Use Silent Bluetooth for distant Server.
- Update some CSS

stevejcl committed:
Update server.ts

detect Bluetooth is a Post Request!

stevejcl committed:
CSS for phone for Camera Page

Add translation for OBS

stevejcl committed:
Update cameras.tsx

error translations

stevejcl committed:
Update CameraTimeLapseSettings.tsx

Correction Time Lapse Settings depend on Dwarf Type!

stevejcl committed:
Add specific css for modall

To avoid conflict with others modals

stevejcl committed:
import Telescopius Today List

stevejcl committed:
compilation Error corrected

stevejcl committed:
Update ImportObservationListModal.tsx

stevejcl committed:
Update start.sh

Unix Format for Docker

stevejcl committed:
Update README.md

Add Prox documentation

stevejcl committed:
Create Proxy-Settings.png

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

stevejcl committed:
Update README.md

stevejcl committed:
Dual Certifcates version

Use a CA certificate for Dwarfium
Use a different certificate for Server and Proxy if using in different computer.

stevejcl committed:
add specific build architecture

Correction for building  DwarfProxy and CreateCert for Linux and Mac
Add ARM 64 build in github workflow : to be tested

stevejcl committed:
Update README.md

stevejcl committed:
Update build.yml

stevejcl committed:
Update build.yml

stevejcl committed:
Update README.md

stevejcl committed:
Add Screenshot for Read Me

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

stevejcl committed:
update images

stevejcl committed:
Update README.md

stevejcl committed:
Update SlaveMode_App-3.jpg

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

stevejcl committed:
Update README.md

stevejcl committed:
Add Personal Library

Correction for Proxy - MediaMtx

stevejcl committed:
Update Translations

stevejcl committed:
add css for mobile and session name

stevejcl committed:
format fix

stevejcl committed:
Restore Bootstrap for larger screen

refresh camera
avoid confict in css too
use proxy for axios request

stevejcl committed:
Repair compil Errors

stevejcl committed:
Update README.md

stevejcl committed:
Create UnlockHostMode.png

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

stevejcl committed:
Update README.md

stevejcl committed:
Merge branch 'apiV2' of https://github.com/stevejcl/dwarfium into apiV2

stevejcl committed:
correction for automatic start
